infra/src/test/kotlin/fi/fta/geoviite/infra/ui/testgroup1/UserRoleTestUI.kt

```diff
@@ -68,30 +104,80 @@ fun assertPublicationLog(frontpage: E2EFrontPage) {
         .returnToFrontPage()
 }
 
-fun assertMapPage(role: E2ERole) {
+fun assertMapPage(role: E2ERole, trackNumber: TrackNumber, designName: String) {
     val mapPage = E2EAppBar().goToMap()
 
     when (role) {
         E2ERole.Operator,
         -> {
             mapPage
+                .also { assertDraftAndDesignModeTabsVisible() }
+                // TODO: Design mode is checked before draft mode because for whatever reason design mode selections fail
+                // if draft mode checks go before them. Find out why
+                .switchToDesignMode()
+                .also { it.toolBar.workspaceDropdown().selectByName(designName) }
+                .also { assertTrackLayoutPageEditButtonsVisible(it, trackNumber) }
                 .switchToDraftMode()
+                .also { assertTrackLayoutPageEditButtonsVisible(it, trackNumber) }
                 .goToPreview()
                 .waitForAllTableValidationsToComplete()
                 .goToTrackLayout()
                 .switchToOfficialMode()
         }
 
-        E2ERole.Team,
+        E2ERole.Team -> {
+            mapPage
+                .also { assertDraftAndDesignModeTabsVisible() }
```

Comment:
Näissä voisi tietysti käyttää myös sitä funktioreferenssikutsua `.also(::assertDraftAndDesignModeTabsVisible)`. Tosin se on vähän formatoinnin puolesta ankea, että whitespace yms on näissä erilainen. Ei sinänsä siis ainakaan mulle väliä kumpi käytössä.

Sen sijaan itse funkkari olisi voinut olla jaettu kahdeksikin, nythän sillä on kaksi eri roolia. Toki ehkä vähän typerää, koska nämä kuitenkin tarkistellaan yhdessä toistaiseksi, mutta jotenkin kiinnittyi tuohon huomiota.

Reply:
Joo, mietin itsekin aika pitkään että miten nämä kannattaisi funktioida. Päädyin nykyiseen ratkaisuun sillä, että nämä visibilityt halutaan tarkastella juuri näin useammalla eri roolilla. Siihen käyttöön tuntui fiksuimmalta pistää molemmat tarkastelut samaan funktioon

Reply:
Itse suosin yleensä aina tuota `::funkkari` muotoa missä se sopii, mutta se on kyllä lähinnä javasta opittu käytäntö (siellä säästi merkkikohinaa oleellisemmin kuin kotlinissa). Tässä en sitä käyttäisi koska silloin tuosta erehtyy luulemaan että tuo assert oikeasti käyttää mapPagea johonkin. Tästä sentään näkee että tuo funkkari ei liity siihen mitenkään.